clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1102,6 +1135,9 @@ private Record nextFetchedRecord() {
                     bytesRead += record.sizeInBytes();
                     nextFetchOffset = record.offset() + 1;
                 }
+                // In some cases, the deserialization may have thrown an exception and the retry may succeed,
```

Comment:
Did you have a specific use case in mind where this was possible? Typically deserializers are deterministic.